### ISSUE TITLE
cdn.onesignal.com blocked by adblocking software(browser extensions)

### ISSUE THREAD

**Issue by anton-piliugin** [OneSignal/OneSignal-Website-SDK#371]
This is a critical issue as people can't even subscribe. May be you'll try to contact the list owners or/and ad blocking software developers. Thanks!

**Comment by tixastronauta** [comment]
What extension? I tried to replicate with ABP (AdBlock Plus), and fortunately it did not block requests to cdn.onesignal.com.

**Comment by anton-piliugin** [comment]
ublock, they have list called "Peter Lowe’s Ad and tracking server list" and i think it's because of it

**Comment by jkasten2** [comment]
Thanks for letting us know. It looks like a few other push provides are blocked too in their list too. There isn't much we can do in this case if they choose to block the OneSignal domain. I recommend asking uBlock if they can remove cdn.onesignal.com as it is an important tool you use to provide updates to your users. In the end though it will be their decision.

**Comment by AndrewMorsillo** [comment]
@jkasten2 is it OK to simply download the SDK from cdn.onesignal.com and host it on my own server so it does not get blocked?

**Comment by jkasten2** [comment]
@AndrewMorsillo This is ok to do but OneSignal won't be able to provide support if your hosting an older version so make sure to check for updates regularly.
https://github.com/OneSignal/OneSignal-Website-SDK/releases

It is also possible breaking changes could be made to the OneSignal service worker (OneSignalSDKWorker.js). Since this gets updated in the browser from the cdn.onesignal.com when a user receives a new push payload we don't test old implements here so it is very important to keep this updated on your end.

**Comment by sido420** [comment]
cdn.onesignal.com is also mentioned at https://github.com/StevenBlack/hosts/issues/860 

Is there anyway we could figure out if cdn.onesignal.com is blocked on client's browser and fallback to using our own server hosted file?

**Comment by anton-piliugin** [comment]
> cdn.onesignal.com is also mentioned at [StevenBlack/hosts#860](https://github.com/StevenBlack/hosts/issues/860)
> 
> Is there anyway we could figure out if cdn.onesignal.com is blocked on client's browser and fallback to using our own server hosted file?

Add something like this:
```
if (typeof window.OneSignal === 'function') {
// Not blocked
} else {
// Blocked
}
```

**Comment by sido420** [comment]
But if that domain is blocked on user's machine then how would the script communicate with Onesignal's servers (if it uses cdn.onesignal.com)? 
Can you make sure OneSignalSDK.js never uses cdn.onesignal.com to communicate with server? I dont see a reason why but just wanted to be doubly sure.

Also, we have the other JS files https://documentation.onesignal.com/docs/web-push-custom-code-setup#section--span-class-step-step-3-span-upload-onesignal-sdk. Do they need to be modified in that case? All they do it seems is importing that script from cdn.onesignal.com.

**Comment by anton-piliugin** [comment]
> But if that domain is blocked on user's machine then how would the script communicate with Onesignal's servers (if it uses cdn.onesignal.com)?
> Can you make sure OneSignalSDK.js never uses cdn.onesignal.com to communicate with server? I dont see a reason why but just wanted to be doubly sure.
> 
> Also, we have the other JS files https://documentation.onesignal.com/docs/web-push-custom-code-setup#section--span-class-step-step-3-span-upload-onesignal-sdk. Do they need to be modified in that case? All they do it seems is importing that script from cdn.onesignal.com.

I think you better open a new issue instead of asking here, because this issue is closed by OneSignal developers

**Comment by verificatorrus** [comment]
Hello everyone! I faced this problem too and have a working solution. I created a proxy for CloudFlare workers.
https://github.com/verificatorrus/onesignal-proxy
